pandas/_libs/lib.pyx

```diff
@@ -783,15 +785,15 @@ def get_level_sorter(ndarray[int64_t, ndim=1] label,
 
     for i in range(len(starts) - 1):
         l, r = starts[i], starts[i + 1]
-        out[l:r] = l + label[l:r].argsort(kind='mergesort')
+        out[l:r] = l + np.asarray(label)[l:r].argsort(kind='mergesort')
```

Comment:
This is being called inside a loop, may need to check perf

Reply:
Moved creation of ndarray to outside the loop.